### PR TITLE
Add request options to type definitions

### PR DESCRIPTION
**Pull request by billytrend**
I need to fully check I picked the right type here but it looks like it would address #1377

### CODE REVIEW

**Comment by wolfy1339** [comment]
I would have personally tested this myself, but since `node-gyp` doesn't support VS 2019 and npm doesn't support NodeJS 12 yet, I cannot.
I'm going to leave this for other contributors

File: scripts/templates/index.d.ts.tpl [hunk 1]
```diff
@@ -5,6 +5,7 @@
 /// <reference lib="esnext.asynciterable" />
 
 import * as http from 'http'
+import { OctokitRequestOptions } from "@octokit/request/dist-types/types";
```
Comment (gr2m) [inline scripts/templates/index.d.ts.tpl:8]: If we depend on internal APIs we have to pin the `@octokit/request` version in package.json

**Comment by gr2m** [comment]
@billytrend can you share a code example that currently does not work with `@octokit/rest` but would work once your PR is merged?

**Comment by wolfy1339** [comment]
@billytrend Are you still interested in taking up this PR?

**Comment by wolfy1339** [comment]
@billytrend ping

**Comment by gr2m** [comment]
I would suggest to just copy & paste the definitions for now, I did that in a few other libraries, too.

The proper fix would be to have one repository with all the type definitions, e.g. `octokit/common-types.ts`, at least for the type definitions that are shared across multiple libraries

**Review by ArtekArtek — APPROVED** [review]

**Comment by gr2m** [comment]
https://github.com/octokit/types.ts/ is a thing now

**Comment by gr2m** [comment]
I think this is obsolete now?

**Pull request closed without merging**